plugins/chat/app/services/chat/create_direct_message_channel.rb

```diff
@@ -0,0 +1,120 @@
+
+    # @!method call(guardian:, **params_to_create)
+    #   @param [Guardian] guardian
+    #   @param [Hash] params_to_create
+    #   @option params_to_create [Array<String>] target_usernames
+    #   @return [Service::Base::Context]
+
+    policy :can_create_direct_message
+    contract
+    model :target_users
+    policy :does_not_exceed_max_direct_message_users,
+           class_name: Chat::DirectMessageChannel::MaxUsersExcessPolicy
+    model :user_comm_screener
+    policy :acting_user_not_disallowing_all_messages
+    policy :acting_user_can_communicate_with_all_parties,
+           class_name: Chat::DirectMessageChannel::CanCommunicateAllPartiesPolicy
+    model :direct_message, :fetch_or_create_direct_message
+    model :channel, :fetch_or_create_channel
+    step :update_memberships
+    step :publish_channel
+
+    # @!visibility private
+    class Contract
+      attribute :target_usernames
+
+      before_validation do
+        self.target_usernames =
+          (
+            if target_usernames.is_a?(String)
+              target_usernames.split(",")
+            else
+              target_usernames
+            end
+          )
+      end
+
+      validates :target_usernames, presence: true, length: { minimum: 1 }
+    end
+
+    private
+
+    def can_create_direct_message(guardian:, **)
+      guardian.can_create_direct_message?
+    end
+
+    def fetch_target_users(guardian:, contract:, **)
+      User.where(username: [guardian.user.username, *contract.target_usernames]).to_a
+    end
+
+    def fetch_user_comm_screener(target_users:, guardian:, **)
+      UserCommScreener.new(acting_user: guardian.user, target_user_ids: target_users.map(&:id))
+    end
+
+    def acting_user_not_disallowing_all_messages(user_comm_screener:, **)
+      !user_comm_screener.actor_disallowing_all_pms?
+    end
+
+    def fetch_or_create_direct_message(target_users:, **)
+      Chat::DirectMessage.for_user_ids(target_users.map(&:id)) ||
+        Chat::DirectMessage.create(user_ids: target_users.map(&:id))
+    end
+
+    def fetch_or_create_channel(direct_message:, **)
+      Chat::DirectMessageChannel.find_or_create_by(chatable: direct_message)
+    end
+
+    def update_memberships(guardian:, channel:, target_users:, **)
+      always_level = Chat::UserChatChannelMembership::NOTIFICATION_LEVELS[:always]
+      # sql_params = {
+      #   acting_user_id: guardian.user.id,
+      #   user_ids: target_users.map(&:id) + [guardian.user.id],
+      #   chat_channel_id: channel.id,
+      #   always_notification_level: Chat::UserChatChannelMembership::NOTIFICATION_LEVELS[:always],
+      # }
+
+      memberships =
+        target_users.map do |user|
+          {
+            user_id: user.id,
+            chat_channel_id: channel.id,
+            muted: false,
+            following: true,
+            desktop_notification_level: always_level,
+            mobile_notification_level: always_level,
+            created_at: Time.zone.now,
```

Comment:
There’s a shortcut to write this, `Time.current` :grin:

Reply:
Didn't know that, but we use `Time.zone.now` way more often in the codebase so I would rather it be consistent

Reply:
Won't die on this hill, but really I dont think using `Time.current` could be classified as being inconsistent. It's also something you can grep/replace in a commit if that is an issue.

Reply:
I think at some point Rubocop was suggesting to use `Time.current` but it seems both forms are equally accepted (`Time.current` is just shorter). That’s basically what is said in the Rails style guide: https://rails.rubystyle.guide/#time-now :sweat_smile:

Reply:
`Time.current` has 16 results in core, `Time.zone.now` has 361, so I would class that as inconsistent :) Leaving it as is